core/src/main/scala/dissonance/data/InteractionType.scala

```diff
@@ -1,16 +1,28 @@
 package dissonance.data
 
-import io.circe.Decoder
+import cats.syntax.all._
+import enumeratum.{Enum, EnumEntry}
+import io.circe.{Decoder, Encoder}
 
-sealed trait InteractionType extends Product with Serializable
+sealed trait InteractionType extends EnumEntry with Product with Serializable
 
-object InteractionType {
+object InteractionType extends Enum[InteractionType] {
   case object Ping               extends InteractionType
   case object ApplicationCommand extends InteractionType
 
-  implicit val targetUserTypeDecoder: Decoder[InteractionType] = Decoder[Int].emap {
-    case 1     => Right(Ping)
-    case 2     => Right(ApplicationCommand)
-    case other => Left(s"Unknown interaction type ID: $other")
+  override def values: IndexedSeq[InteractionType] = findValues
```

Comment:
The build failed, I believe you need `import scala.collection.immutable.IndexedSeq` to fix it.